### PR TITLE
MRG fix for #3451

### LINKED ISSUE #3451: ica.plot_overlay - red and black lines are now mixed
[`plot_overlay` mixes red and black lines](http://www.martinos.org/mne/dev/_images/sphx_glr_plot_artifacts_correction_ica_010.png) because `mne.viz.evoked._plot_evoked` uses `z_order` param now and the default 'unsorted' value gives different z-order to each line. When two evokeds are plotted successively (before ICA - red; after ICA - black) this causes the lines to mix.
Potential solutions:
* `_plot_evoked` could expose another z_order option ('no', 'none', same', 'flat'?) that does not set line z-order
* `_plot_evoked` 'unsorted' z_order could lead to not changing z-order at all
* `ica.plot_overlay` could use the option to pass a callable to z_order where the callable would just return `np.ones(n_channels)`

I think the first and second solutions are more general and thus better.

### PR DESCRIPTION
**Pull request by mmagnuski**
solution suggested by @jona-sassenhagen
closes #3451

### CODE REVIEW

**Comment by mmagnuski** [comment]
The biggest drawback of merging this would be that I'll stop having `666` in lines added to mne ;)

File: mne/viz/evoked.py [hunk 1]
```diff
@@ -345,7 +345,7 @@ def _plot_evoked(evoked, picks, exclude, unit, show,
                     # plot channels
                     for ch_idx, z in enumerate(z_ord):
                         line_list.append(ax.plot(times, D[ch_idx], picker=3.,
-                                                 zorder=1 + z,
+                                                 zorder=z + 1 if spatial_colors else 1,
```
Comment (Eric89GXL) [inline mne/viz/evoked.py]: I think by default lines get `zorder=2` in matplotlib, can you check? Or do you mean to set the zorder lower?

Comment (jona-sassenhagen) [inline mne/viz/evoked.py]: It used to be 1:

https://github.com/mne-tools/mne-python/blob/b5d5194151fa12e839a55495932fdbfe81d73795/mne/viz/evoked.py#L316

But maybe it should be the MPL default. It should definitely be in front of spines though.

Comment (mmagnuski) [inline mne/viz/evoked.py]: Yes, I've checked, the default zorder is 2, I'm changing to `2 + z ...`

**Comment by agramfort** [comment]
@jona-sassenhagen happy?

File: mne/viz/evoked.py [hunk 2]
```diff
@@ -344,9 +344,10 @@ def _plot_evoked(evoked, picks, exclude, unit, show,
 
                     # plot channels
                     for ch_idx, z in enumerate(z_ord):
-                        line_list.append(ax.plot(times, D[ch_idx], picker=3.,
-                                                 zorder=1 + z,
-                                                 color=colors[ch_idx])[0])
+                        line_list.append(
+                            ax.plot(times, D[ch_idx], picker=3.,
+                                    zorder=z + 1 if spatial_colors else 1,
+                                    color=colors[ch_idx])[0])
```
Comment (jona-sassenhagen) [inline mne/viz/evoked.py]: Could also save a line by

```
                    for ch_idx, z in enumerate(z_ord):
                        line_list.append(ax.plot(
                            times, D[ch_idx], picker=3., color=colors[ch_idx],
                            zorder=z + 1 if spatial_colors else 1)[0])
```
but maybe not worth it

**Comment by jona-sassenhagen** [comment]
So you checked the (non-spatial colors) plot and it now works like before?

If yes, merge

**Comment by agramfort** [comment]
@mmagnuski ready for merge?

**Comment by mmagnuski** [comment]
Yes, checked `plot_overlay` and it looks good. 
Hyperdrive activated... ready for merge! :rocket:

File: mne/viz/evoked.py (hunk 2 above)
Comment (mmagnuski) [inline mne/viz/evoked.py]: right, but should I deactivate hyperdrive and change it? :) 
I think I'll just leave it the way it is, looks ok to me.

**Pull request merged**